services/121-service/test/registrations/pagination/filter-registration-range.test.ts

```diff
@@ -0,0 +1,281 @@
+import { FilterOperator } from 'nestjs-paginate';
+
+import { SeedScript } from '@121-service/src/scripts/enum/seed-script.enum';
+import {
+  getRegistrations,
+  seedIncludedRegistrations,
+} from '@121-service/test/helpers/registration.helper';
+import {
+  getAccessToken,
+  resetDB,
+} from '@121-service/test/helpers/utility.helper';
+import {
+  programIdOCW,
+  registrationOCW1,
+  registrationOCW2,
+  registrationOCW3,
+  registrationOCW4,
+  registrationOCW5,
+} from '@121-service/test/registrations/pagination/pagination-data';
+
+const registrations = [
+  registrationOCW1,
+  registrationOCW2,
+  registrationOCW3,
+  registrationOCW4,
+  registrationOCW5,
+];
+const allReferenceIds = registrations.map(
+  (registration) => registration.referenceId,
+);
+
+async function getMiddleDateOfRegistration(accessToken: string): Promise<Date> {
+  return getRegistrations({
+    programId: programIdOCW,
+    accessToken,
+  }).then((response) => {
+    const registrations = response.body.data;
+    const sortedCreatedDates = registrations
+      .map((registration) => new Date(registration.created))
+      .sort((a, b) => a.getTime() - b.getTime());
+
+    // Pick the middle date dynamically and adjust by subtracting a few milliseconds
+    const middleIndex = Math.floor(sortedCreatedDates.length / 2);
+    const middleCreatedDate = new Date(sortedCreatedDates[middleIndex]);
+    middleCreatedDate.setMilliseconds(middleCreatedDate.getMilliseconds() - 5);
+
+    return middleCreatedDate;
+  });
+}
+
+describe('Filter registrations using', () => {
+  let accessToken: string;
+  let middleCreatedDate: Date;
+
+  beforeAll(async () => {
+    await resetDB(SeedScript.nlrcMultiple);
+    accessToken = await getAccessToken();
+
+    await seedIncludedRegistrations(
+      registrations.slice(0, 2),
+      programIdOCW,
+      accessToken,
+    );
+    await seedIncludedRegistrations(
+      registrations.slice(2, 3),
+      programIdOCW,
+      accessToken,
+    );
+    await seedIncludedRegistrations(
+      registrations.slice(3, 5),
+      programIdOCW,
+      accessToken,
+    );
+    middleCreatedDate = await getMiddleDateOfRegistration(accessToken);
+  });
+
+  it('should filter registration root numeric attributes based on greater than', async () => {
```

Comment:
This description is too cryptic for me. Does it do something with square roots of numeric values?